src/Device/deviceActions.js

```diff
@@ -235,7 +235,7 @@ export const stopWatchingDevices = () => {
  * @param {Array<String>} [choices] The choices to display to the user (optional).
  * @returns {Promise<String>} Promise that resolves with the user input.
  */
-const getDeviceSetupUserInput = (dispatch, message, choices) => new Promise((resolve, reject) => {
+const getDeviceSetupUserInput = (message, choices) => dispatch => new Promise((resolve, reject) => {
```

Comment:
This must be the other way around (you already bind the `dispatch` when calling it once and leave the other parameters to be bound later: 
```suggestion
const getDeviceSetupUserInput = dispatch => (message, choices) => new Promise((resolve, reject) => {
```